ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+        param_to_column_mapping = all_schemas_parameters[parser_file['Normalization']['Schema']]
+        for param in parser_file['ParserParams']:
+            with self.subTest():
+                if param['Name'] not in param_to_column_mapping:
+                    self.fail(f"parameter: {param['Name']} - is not a valid parameter")
+                column_name_in_table = param_to_column_mapping[param['Name']]
+                self.handle_param(param, no_call_query, columns_in_answer, column_name_in_table)            
+                      
+    # Sending a parameters to the suitable test according to the parameter type    
+    def handle_param(self, param, no_call_query, columns_in_answer, column_name_in_table):
+        param_name = param['Name']
+        param_type = param['Type']
+        if (param_name == "pack"):
+            return 
+        if (param_name == "disabled"):
+            self.disabled_test(param, no_call_query)
+        elif  column_name_in_table not in columns_in_answer:
+            return
+        elif (param_type == "datetime"):
+            pass
+        elif (param_type == "dynamic"):
+            pass
+        else:
+            self.scalar_test(param, no_call_query, column_name_in_table)
+
+    
+    # Checking if the provided workspace has some data
+    def check_data_in_workspace(self, no_call_query):
+        check_data_response = self.send_query(no_call_query + "query() | take 5")
+        if len(check_data_response.tables[0].rows) == 0:
+            self.fail("No data in the provided workspace")
+          
+
+    # Checking if all fields from "required_fields" array are in the yaml file.
+    def check_required_fields(self, parser_file):
+        required_fields = ["ParserParams", "ParserQuery", "Normalization.Schema" ]
+        missing_fields = []
+        for full_field in required_fields:
+            file = parser_file
+            fields = full_field.split('.')
+            for field_name in fields:
+                if field_name not in file:
+                    missing_fields.append(full_field)
+                    break
+                file = file[field_name]
+        if len(missing_fields) != 0:
+            self.fail(f"The following fields are missing in the file:\n{missing_fields}")
+    
+
+    # Test for parameter which are not datetime,dynamic or disabled
+    def scalar_test(self, param, no_call_query, column_name_in_table):
+        param_name = param['Name']
+        no_filter_query = no_call_query + create_call_without_parameter(column_name_in_table)
+        no_filter_response = self.send_query(no_filter_query)
+        self.assertNotEqual(len(no_filter_response.tables[0].rows) , 0 , f"No data for parameter:{param_name}")
+        with  self.subTest():
+            self.assertNotEqual(len(no_filter_response.tables[0].rows), 1, f"Only one value exists for parameter: {param_name} - validations for this parameter are partial" )
```

Comment:
I'd put it in a different function and call it here

Reply:
Preferred not to move this part to an other function as the response to the query is needed to set up the next query call (the call in "scalar_test_check_filtering")